functions/consumer/rabbit-consumer/src/main/java/org/springframework/cloud/fn/consumer/rabbit/RabbitConsumerConfiguration.java

```diff
@@ -144,25 +146,49 @@ public void destroy() {
 		}
 	}
 
-	private static class AutoConfig extends RabbitAutoConfiguration {
-
-		static class Creator extends RabbitConnectionFactoryCreator {
-
-//			@Override
-			public CachingConnectionFactory rabbitConnectionFactory(CachingConnectionFactory cachingConnectionFactory)
-					throws Exception {
-//				CachingConnectionFactory cf = super.rabbitConnectionFactory(config, resourceLoader, credentialsProvider,
-//						credentialsRefreshService, connectionNameStrategy, connectionFactoryCustomizers);
-				cachingConnectionFactory.setConnectionNameStrategy(
-						connectionFactory -> "rabbit.sink.own.connection");
-				cachingConnectionFactory.afterPropertiesSet();
-				return cachingConnectionFactory;
-			}
-
-		}
+	private ConnectionFactory buildLocalConnectionFactory() throws Exception {
+		this.ownConnectionFactory = rabbitConnectionFactory(this.bootProperties, this.resourceLoader,
+				this.credentialsProvider, this.credentialsRefreshService, this.connectionFactoryCustomizers);
+		return this.ownConnectionFactory;
 	}
 
-}
-
-
+	/**
+	 *
+	 * @param properties
+	 * @param resourceLoader
+	 * @param credentialsProvider
+	 * @param credentialsRefreshService
+	 * @param connectionFactoryCustomizers
+	 * @return
+	 * @throws Exception
+	 */
+	private CachingConnectionFactory rabbitConnectionFactory(RabbitProperties properties, ResourceLoader resourceLoader,
+			ObjectProvider<CredentialsProvider> credentialsProvider,
+			ObjectProvider<CredentialsRefreshService> credentialsRefreshService,
+			ObjectProvider<ConnectionFactoryCustomizer> connectionFactoryCustomizers) throws Exception {
+
+		/* NOTE: This is based on RabbitAutoConfiguration.RabbitConnectionFactoryCreator
+		 * 		https://github.com/spring-projects/spring-boot/blob/c820ad01a108d419d8548265b8a34ed7c5591f7c/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAutoConfiguration.java#L95
+		 * [UPGRADE_CONSIDERATION] this should stay somewhat in sync w/ the functionality provided by its original source.
+		 */
+		RabbitConnectionFactoryBean connectionFactoryBean = new RabbitConnectionFactoryBean();
```

Comment:
Because this code is identical in the supplier and consumer I was thinking we could move it somewhere common and share it between the 2.

Reply:
Right. For other similar we have a dedicated module in the `common`, e.g. `mqtt-common`, `aws-s3-common` etc.

We may have `rabbit-common` for this "common" code, but does it really justify already a whole module?

(Let's revise it separately!)